### ISSUE TITLE
`not in` query does not work with numeric values

### ISSUE THREAD

**Issue by kolaente** [typesense/typesense#2038]
## Description

A schema with a field like this:

```
{"name": "other_id", "type": "int64" }
```

Filtering with a query like `other_id:![1]` will result in an error `Error with filter field `other_id`: Numerical field has an invalid comparator.`.

At the same time, filtering for `other_id:[1]` works just fine.

I would expect this to be consistent between `in` and `not in` queries.

## Steps to reproduce

1. Create a collection with a field like above
2. Add some documents
3. Try to filter

## Metadata

**Typesense Version**: 27.1

**OS**: official docker image

**Comment by happy-san** [comment]
@kolaente Can you try again with `other_id:!=[1]` ?

**Comment by kolaente** [comment]
That seems to work. So this is only a docs issue? I got the operator from [this page](https://typesense.org/docs/guide/tips-for-filtering.html#available-operators)

**Comment by happy-san** [comment]
Yeah, it's a docs issue. [PR for fix](https://github.com/typesense/typesense-website/pull/277)

**Comment by tharropoulos** [comment]
@vladjerca I've posted a PR on #2474 to support the filter syntax on all cases (numbers, strings, booleans)
